support multinomial for dynamic num_samples

avikchaudhuri opened the pull request:
Test Plan: added test

Fixes #149048

Differential Revision: D71434914

albanD commented on test/export/test_export.py:
```diff
@@ -6832,6 +6832,24 @@ def forward(self, x, y):
+        ep = torch.export.export(
+            model, inputs, dynamic_shapes={"x": {0: DYNAMIC, 1: DYNAMIC}, "y": {0: DYNAMIC}}
+        )
+        print(ep)
```
This is not really a test?

avikchaudhuri replied:
yeah gonna fix this up

pianpwk reviewed (approved):
niceeeeee

avikchaudhuri committed:
support multinomial for dynamic num_samples (#149463)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/149463

Test Plan: added test

Reviewed By: pianpwk

Differential Revision: D71434914
